### PR TITLE
Ensure txn-id is incremental using sqlalchemy

### PR DESCRIPTION
**Pull request by AbdealiLoKo**
In sqlalchemy, lock the DB and ensure that the txn-id will always be incrementing.
This is an assumption that is baked into the library and so we should ensure this assumption never breaks for txn-id atleast.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -27,6 +27,8 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Black Validation
         uses: psf/black@stable
+        with:
+          version: "22.8.0"  # Last version which can be used in py3.6 and py3.7
```
Comment (shivamv300) [inline .github/workflows/main.yml:31]: are we supporting py3.6 anymore?
can we use a newer version which supports py3.7? (23.3.0)

File: sqlalchemy_history/unit_of_work.py [hunk 2]
```diff
@@ -113,6 +113,18 @@ def create_transaction(self, session):
...
+        # Fetch the last-id in Transaction table
+        # We are manually setting the transaction ID - to guarantee that the transaction ID is always
+        # increasing. In some cases like Oracle RAC - the order of the IDs generated are not sequential.
+        # Ref: https://stackoverflow.com/questions/57620096
+        # So, here we ourselves are setting the transaction IDs using table locks
+
+        # FIXME: There is a small race condition here when 2 APIs could write concurrently between the
+        #        last_transaction_id fetching and FLUSH
+        last_transaction_id = self.version_session.query(sa.func.max(Transaction.id)).scalar()
+        self.current_transaction.id = last_transaction_id
```
Comment (shivamv300) [inline sqlalchemy_history/unit_of_work.py:126]: should we not be doing `last_txn_id + 1` ?

File: .github/workflows/main.yml (hunk 1 above)
Comment (AbdealiLoKo) [reply .github/workflows/main.yml:31]: 22.8.0 is the last version for 3.6 and 3.7
After that they only have 3.8 support

File: sqlalchemy_history/unit_of_work.py (hunk 2 above)
Comment (AbdealiLoKo) [reply sqlalchemy_history/unit_of_work.py:126]: Yes -_- will fix

**Review by shivamv300 — APPROVED** [review]

**Comment by AbdealiLoKo** [comment]
This causes race conditions where if 2 API calls come very quickly - the second one gives an error.

**Pull request closed without merging**